File: fixtures/scale/objc/large/Families/Atbash/AtbashDescriptor.m
#import "AtbashDescriptor.h"
#import "AtbashCipher.h"
#import "AtbashVectors.h"

@implementation AtbashDescriptor

- (NSString *)family {
    return @"atbash";
}

- (NSString *)suite {
    return @"classical";
}

- (id<CLCipher>)cipher {
    return [[AtbashCipher alloc] init];
}

- (NSArray *)vectors {
    return [AtbashVectors all];
}

@end

File: fixtures/scale/php/large/src/Families/Atbash/AtbashDescriptor.php
<?php

namespace CipherLab\Families\Atbash;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together AtbashCipher and
// AtbashVectors under the "classical" suite.
class AtbashDescriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'atbash';
    }

    public function suite(): string
    {
        return 'classical';
    }

    public function cipher(): Cipher
    {
        return new AtbashCipher();
    }

    public function vectors(): array
    {
        return AtbashVectors::all();
    }
}
